src/hotspot/share/opto/library_call.cpp

```diff
@@ -6848,3 +6851,20 @@ bool LibraryCallKit::inline_getObjectSize() {
 
   return true;
 }
+
+//------------------------------- inline_blackhole --------------------------------------
+//
+// Make sure all arguments to this node are alive.
+// This matches methods that were requested to be blackholed through compile commands.
+//
+bool LibraryCallKit::inline_blackhole() {
```

Comment:
I can't see where receiver null check comes from. Don't you have to explicitly insert it in the intrinsic?

Reply:
Dang. We should do receiver null-checks in intrinsics. Added. It looks like C2 is able to elide null-checks in JMH hot loops, while C1 leaves the first null-check in subsequent `Blackhole` calls.